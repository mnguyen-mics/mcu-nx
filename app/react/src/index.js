import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
