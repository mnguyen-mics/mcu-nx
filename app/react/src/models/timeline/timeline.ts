import { Index } from '../../utils';

export interface Activity {
  $email_hash: string | object;
  $events: EventProps[];
  $location: {
    $latlon: string[];
  };
  $origin: object;
  $session_duration: number;
  $session_status:
    | 'IN_SESSION'
    | 'SESSION_SNAPSHOT'
    | 'CLOSED_SESSION'
    | 'NO_SESSION';
  $site_id: string;
  $topics: Index<Index<number>>;
  $ts: number;
  $ttl: number;
  $type: string;
  $user_account_id: string;
  $user_agent_id: string;
  $app_id: string;
  $node_id?: number;
  $node_name?: string;
  $scenario_name?: string;
  $scenario_id?: number;
  $previous_node_id?: number;
  $previous_node_name?: string;
}

export interface ActivityCardProps {
  activity: Activity;
  datamartId: string;
  identifiers: IdentifiersProps;
}

export interface UserScenarioActivityCardProps {
  activity: Activity;
}

export interface Property {
  $referrer: string;
  $url: string;
  language: string;
}

export interface EventProps {
  $event_name: string;
  $properties: Property[];
  $ts: number;
}

export interface UserAgent {
  creation_ts: number;
  device: Device;
  last_activity_ts: number;
  mappings: any[]; // type it better
  providers: any[]; // type it better
  type: 'USER_AGENT';
  vector_id: string;
}

export interface UserPoint {
  type: 'USER_POINT';
  user_point_id: string;
}

export type UserIdentifierShape = UserAgent | UserPoint;

export type FormFactor = 'TABLET' | 'SMARTPHONE' | 'PERSONAL_COMPUTER';

export interface Device {
  brand?: string;
  browser_family?: string;
  browser_version?: string;
  carrier?: string;
  form_factor?: FormFactor;
  model?: string;
  os_family?: string;
  os_version?: string;
  raw_value?: string;
}

export interface IdentifiersProps {
  isLoading: boolean;
  hasItems: boolean;
  items: {
    USER_ACCOUNT: any[]; // type it better
    USER_AGENT: UserAgent[];
    USER_EMAIL: any[]; // type it better
    USER_POINT: UserPoint[];
  };
}

export interface UserProfileResource {
  creation_date?: number;
  birth_date?: number;
  first_name?: string;
  last_name?: string;
  email?: string;
  isClient?: boolean;
}

export interface UserSegmentResource {
  segment_id: string;
  data_bag: string;
  last_modified_ts?: number;
  creation_ts?: number;
  expiration_ts?: number;
}

export interface Cookies {
  mics_lts?: string;
  mics_uaid?: string;
  mics_vid?: string;
}

export interface OriginProps {
  $campaign_id?: string;
  $campaign_name?: string;
  $campaign_technical_name?: string;
  $channel?: string;
  $creative_id?: string;
  $creative_name?: string;
  $creative_technical_name?: string;
  $engagement_content_id?: string;
  $gclid?: string;
  $keywords?: string;
  $log_id?: string;
  $message_id?: string;
  $message_technical_name?: string;
  $referral_path?: string;
  $social_network?: string;
  $source?: string;
  $sub_campaign_id?: string;
  $sub_campaign_technical_name?: string;
  $ts?: number;
}
