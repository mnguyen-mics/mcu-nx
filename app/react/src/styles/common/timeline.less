.mcs-monitoring {
    .mcs-subtitle {
        font-weight: 500;
        color: darken(@layout-content-background, 30%);
        margin-bottom: 20px;
    }
    .mcs-timeline {
        .mcs-title {
            font-size: 14px;
            color: @primary-color;
            font-weight: 500;
            letter-spacing: 0.5px;
            text-align: left;
            margin-bottom: 20px;
        }
        .ant-timeline-item-head {
            background-color:  @layout-content-background!important;
            .mcs-timeline-dot {
                font-size: 22px;
                color: @info-color!important;
            }
            .mcs-timeline-dot.live {
                font-size: 22px;
                color: @success-color!important;
            }
        }
        .ant-timeline-item-content {
            margin-bottom: 15px;
        }
    }
    .mcs-timeline-last-dot {
        font-size: 22px;
        color: @info-color!important;
    }
    .ant-timeline-item-head {
        background-color: #f1f1f1 !important;
    }
}

