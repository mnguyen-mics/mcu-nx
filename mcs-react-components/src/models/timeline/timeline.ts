import { Index } from '../../utils';
import { UserActivityEventResource } from '../datamart/UserActivityResource';
import {
  OperatingSystemFamily,
  BrowserFamily,
} from '../datamart/graphdb/RuntimeSchema';

export interface Activity {
  $email_hash: string | object;
  $events: UserActivityEventResource[];
  $location: {
    $latlon: string[];
  };
  $origin: object;
  $session_duration: number;
  $session_status:
    | 'IN_SESSION'
    | 'SESSION_SNAPSHOT'
    | 'CLOSED_SESSION'
    | 'NO_SESSION';
  $site_id: string;
  $topics: Index<Index<number>>;
  $ts: number;
  $ttl: number;
  $type: string;
  $user_account_id: string;
  $user_agent_id: string;
  $app_id: string;
  $node_id?: number;
  $node_name?: string;
  $scenario_name?: string;
  $scenario_id?: number;
  $previous_node_id?: number;
  $previous_node_name?: string;
}

export interface UserScenarioActivityCardProps {
  activity: Activity;
}

export interface Property {
  $referrer: string;
  $url: string;
  language: string;
}

export interface UserAgent {
  creation_ts: number;
  device: Device;
  last_activity_ts: number;
  mappings: any[]; // type it better
  providers: any[]; // type it better
  type: 'USER_AGENT';
  vector_id: string;
}

export interface UserEmailIdentifierProviderResource {
  technical_name: string;
  creation_ts?: number;
  last_activity_ts?: number;
  last_modified_ts?: number;
  expiration_ts?: number;
  active?: boolean;
  status?: string;
}

export interface UserAgentInfo {
  form_factor: FormFactor;
  os_family: OperatingSystemFamily;
  browser_family?: BrowserFamily;
  browser_version: string | null;
  brand: string | null;
  model: string | null;
  os_version: string | null;
  carrier: string | null;
  raw_value: string | null;
  agent_type: string | null;
}

export interface UserAgentIdentifierProviderResource {
  technical_name: string;
  creation_ts?: number;
  last_activity_ts?: number;
  expiration_ts?: number;
}

export interface UserAgentIdMappingResource {
  user_agent_id: string;
  realm_name: string;
  last_activity_ts: number;
}

export interface UserPointIdentifierInfo {
  user_point_id: string;
  creation_ts?: number;
  type: 'USER_POINT';
}

export interface UserEmailIdentifierInfo {
  hash: string;
  email?: string;
  operator?: string;
  creation_ts: number;
  last_activity_ts: number;
  providers: UserEmailIdentifierProviderResource[];
  type: 'USER_EMAIL';
}

export interface UserAccountIdentifierInfo {
  user_account_id: string;
  creation_ts: number;
  compartment_id?: number;
  type: 'USER_ACCOUNT';
}

export interface UserAgentIdentifierInfo {
  vector_id: string;
  device?: UserAgentInfo;
  creation_ts: number;
  last_activity_ts: number;
  providers: UserAgentIdentifierProviderResource[];
  mappings: UserAgentIdMappingResource[];
  type: 'USER_AGENT';
}

export type UserIdentifierInfo =
  | UserPointIdentifierInfo
  | UserEmailIdentifierInfo
  | UserAccountIdentifierInfo
  | UserAgentIdentifierInfo;

export function isUserPointIdentifier(
  userIdentifier: UserIdentifierInfo,
): userIdentifier is UserPointIdentifierInfo {
  return userIdentifier.type === 'USER_POINT';
}

export function isUserAccountIdentifier(
  userIdentifier: UserIdentifierInfo,
): userIdentifier is UserAccountIdentifierInfo {
  return userIdentifier.type === 'USER_ACCOUNT';
}

export function isUserAgentIdentifier(
  userIdentifier: UserIdentifierInfo,
): userIdentifier is UserAgentIdentifierInfo {
  return userIdentifier.type === 'USER_AGENT';
}

export function isUserEmailIdentifier(
  userIdentifier: UserIdentifierInfo,
): userIdentifier is UserEmailIdentifierInfo {
  return userIdentifier.type === 'USER_EMAIL';
}

export type FormFactor = 'TABLET' | 'SMARTPHONE' | 'PERSONAL_COMPUTER';

export interface Device {
  brand?: string;
  browser_family?: string;
  browser_version?: string;
  carrier?: string;
  form_factor?: FormFactor;
  model?: string;
  os_family?: string;
  os_version?: string;
  raw_value?: string;
}

export interface IdentifiersProps {
  hasItems: boolean;
  items: {
    USER_ACCOUNT: UserAccountIdentifierInfo[];
    USER_AGENT: UserAgentIdentifierInfo[];
    USER_EMAIL: UserEmailIdentifierInfo[];
    USER_POINT: UserPointIdentifierInfo[];
  };
}

export interface UserProfileResource {
  creation_date?: number;
  birth_date?: number;
  first_name?: string;
  last_name?: string;
  email?: string;
  isClient?: boolean;
}

export interface UserSegmentResource {
  segment_id: string;
  data_bag: string;
  last_modified_ts?: number;
  creation_ts?: number;
  expiration_ts?: number;
}

export interface Cookies {
  mics_lts?: string;
  mics_uaid?: string;
  mics_vid?: string;
}

export interface OriginProps {
  $campaign_id?: string | null;
  $campaign_name?: string | null;
  $campaign_technical_name?: string | null;
  $channel?: string | null;
  $creative_id?: string | null;
  $creative_name?: string | null;
  $creative_technical_name?: string | null;
  $engagement_content_id?: string | null;
  $gclid?: string | null;
  $keywords?: string | null;
  $log_id?: string | null;
  $message_id?: string | null;
  $message_technical_name?: string | null;
  $referral_path?: string | null;
  $social_network?: string | null;
  $source?: string | null;
  $sub_campaign_id?: string | null;
  $sub_campaign_technical_name?: string | null;
  $ts?: number | null;
}
