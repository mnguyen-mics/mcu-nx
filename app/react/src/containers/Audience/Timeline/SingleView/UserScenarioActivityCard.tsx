import * as React from 'react';
import { Row } from 'antd';
import moment from 'moment';
import { compose } from 'recompose';
import { injectIntl, InjectedIntlProps, FormattedMessage } from 'react-intl';
import {
  Activity,
  UserScenarioActivityCardProps,
} from '../../../../models/timeline/timeline';
import { Card } from '../../../../components/Card/index';
import messages from '../messages';
import { RouteComponentProps, withRouter } from 'react-router';
import { TimelinePageParams } from '../TimelinePage';

type Props = UserScenarioActivityCardProps &
  InjectedIntlProps &
  RouteComponentProps<TimelinePageParams>;

interface CardElements {
  title: string,
  intlMsg: React.ReactNode,
}

class UserScenarioActivityCard extends React.Component<Props> {

  generateCardElements(activity: Activity): CardElements {
    switch (activity.$type) {
      case 'USER_SCENARIO_START':
        return {
          title: 'SCENARIO START',
          intlMsg:
            <FormattedMessage {...messages.userScenarioStartContent}
              values={{ scenarioName: <b>{activity.$scenario_name}</b> }}
            />
        }
      case 'USER_SCENARIO_STOP':
        return {
          title: 'SCENARIO STOP',
          intlMsg:
            <FormattedMessage {...messages.userScenarioStopContent}
              values={{ scenarioName: <b>{activity.$scenario_name}</b> }}
            />
        }
      case 'USER_SCENARIO_NODE_ENTER':
        return {
          title: 'SCENARIO NODE ENTER',
          intlMsg:
            <FormattedMessage {...messages.userScenarioNodeEnterContent}
              values={{
                scenarioName: <b>{activity.$scenario_name}</b>,
                scenarioNodeName: <b>{activity.$node_name}</b>
              }}
            />
        }
      case 'USER_SCENARIO_NODE_EXIT':
        return {
          title: 'SCENARIO NODE EXIT',
          intlMsg:
            <FormattedMessage {...messages.userScenarioNodeExitContent}
              values={{
                scenarioName: <b>{activity.$scenario_name}</b>,
                scenarioNodeName: <b>{activity.$node_name}</b>
              }}
            />
        }
      case 'USER_SCENARIO_NODE_MOVEMENT':
        return {
          title: 'SCENARIO NODE MOVEMENT',
          intlMsg:
            <FormattedMessage {...messages.userScenarioNodeMovementContent}
              values={{
                scenarioName: <b>{activity.$scenario_name}</b>,
                scenarioNodeName: <b>{activity.$node_name}</b>,
                scenarioOldNodeName: <b>{activity.$previous_node_name}</b>
              }}
            />
        }
      default:
        return ({
          title: "Scenario item not found",
          intlMsg: "",
        })
    }
  }

  render() {
    const activity = this.props.activity
    const cardContent = this.generateCardElements(activity)
    return (
      <Card title={cardContent.title}>
        <Row>
          <div className="mcs-card-content-text">{cardContent.intlMsg}</div>
        </Row>
        <Row className="border-top sm-footer timed-footer text-right">
          {moment(activity.$ts).format('H:mm:ss')}
        </Row>
      </Card>
    );
  }
}

export default compose<Props, UserScenarioActivityCardProps>(
  injectIntl,
  withRouter,
)(UserScenarioActivityCard);
