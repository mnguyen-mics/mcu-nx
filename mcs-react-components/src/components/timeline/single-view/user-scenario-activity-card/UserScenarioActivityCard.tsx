import * as React from 'react';
import { Row } from 'antd';
import * as _moment from 'moment';
import { compose } from 'recompose';
import { injectIntl, InjectedIntlProps, FormattedMessage, defineMessages } from 'react-intl';
import { Activity, UserScenarioActivityCardProps } from '../../../../models/timeline/timeline';
import Card from '../../../card';

const moment = _moment;

const messages = defineMessages({
  userScenarioStartContent: {
    id: 'audience.monitoring.timeline.activity.scenario.start.content',
    defaultMessage: 'Entered {scenarioName}',
  },
  userScenarioStopContent: {
    id: 'audience.monitoring.timeline.activity.scenario.stop.content',
    defaultMessage: 'Exited {scenarioName}',
  },
  userScenarioNodeEnterContent: {
    id: 'audience.monitoring.timeline.activity.scenario.node.enter.content',
    defaultMessage: 'Entered {scenarioNodeName} node within {scenarioName}',
  },
  userScenarioNodeExitContent: {
    id: 'audience.monitoring.timeline.activity.scenario.node.exit.content',
    defaultMessage: 'Exited {scenarioNodeName} node within {scenarioName}',
  },
  userScenarioNodeMovementContent: {
    id: 'audience.monitoring.timeline.activity.scenario.node.movement.content',
    defaultMessage: 'Moved from {scenarioOldNodeName} to {scenarioNodeName} within {scenarioName}',
  },
});

type Props = UserScenarioActivityCardProps & InjectedIntlProps;

interface CardElements {
  title: string;
  intlMsg: React.ReactNode;
}

class UserScenarioActivityCard extends React.Component<Props> {
  generateCardElements(activity: Activity): CardElements {
    switch (activity.$type) {
      case 'USER_SCENARIO_START':
        return {
          title: 'SCENARIO START',
          intlMsg: (
            <FormattedMessage
              {...messages.userScenarioStartContent}
              values={{ scenarioName: <b>{activity.$scenario_name}</b> }}
            />
          ),
        };
      case 'USER_SCENARIO_STOP':
        return {
          title: 'SCENARIO STOP',
          intlMsg: (
            <FormattedMessage
              {...messages.userScenarioStopContent}
              values={{ scenarioName: <b>{activity.$scenario_name}</b> }}
            />
          ),
        };
      case 'USER_SCENARIO_NODE_ENTER':
        return {
          title: 'SCENARIO NODE ENTER',
          intlMsg: (
            <FormattedMessage
              {...messages.userScenarioNodeEnterContent}
              values={{
                scenarioName: <b>{activity.$scenario_name}</b>,
                scenarioNodeName: <b>{activity.$node_name}</b>,
              }}
            />
          ),
        };
      case 'USER_SCENARIO_NODE_EXIT':
        return {
          title: 'SCENARIO NODE EXIT',
          intlMsg: (
            <FormattedMessage
              {...messages.userScenarioNodeExitContent}
              values={{
                scenarioName: <b>{activity.$scenario_name}</b>,
                scenarioNodeName: <b>{activity.$node_name}</b>,
              }}
            />
          ),
        };
      case 'USER_SCENARIO_NODE_MOVEMENT':
        return {
          title: 'SCENARIO NODE MOVEMENT',
          intlMsg: (
            <FormattedMessage
              {...messages.userScenarioNodeMovementContent}
              values={{
                scenarioName: <b>{activity.$scenario_name}</b>,
                scenarioNodeName: <b>{activity.$node_name}</b>,
                scenarioOldNodeName: <b>{activity.$previous_node_name}</b>,
              }}
            />
          ),
        };
      default:
        return {
          title: 'Scenario item not found',
          intlMsg: '',
        };
    }
  }

  render() {
    const activity = this.props.activity;
    const cardContent = this.generateCardElements(activity);
    return (
      <Card title={cardContent.title} className='mcs-user-scenario-activity-card'>
        <Row>
          <div className='mcs-card-content-text'>{cardContent.intlMsg}</div>
        </Row>
        <Row className='border-top sm-footer timed-footer text-right'>
          {moment(activity.$ts).format('H:mm:ss')}
        </Row>
      </Card>
    );
  }
}

export default compose<Props, UserScenarioActivityCardProps>(injectIntl)(UserScenarioActivityCard);
