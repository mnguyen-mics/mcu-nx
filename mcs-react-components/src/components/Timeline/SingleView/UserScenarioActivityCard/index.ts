import UserScenarioActivityCard from './UserScenarioActivityCard';

export {
  UserScenarioActivityCardProps,
} from '../../../../models/timeline/timeline';

export default UserScenarioActivityCard;
