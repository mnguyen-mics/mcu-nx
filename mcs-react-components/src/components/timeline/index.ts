import Device from './single-view/device';
import DeviceCard from './single-view/device-card';
import EventActivity from './single-view/event-activity';
import Origin from './single-view/origin';
import ProfileInfo from './single-view/profile-info';
import Topics from './single-view/topics';
import UserScenarioActivityCard from './single-view/user-scenario-activity-card';

export default {
  Device,
  DeviceCard,
  EventActivity,
  Origin,
  ProfileInfo,
  Topics,
  UserScenarioActivityCard,
};
