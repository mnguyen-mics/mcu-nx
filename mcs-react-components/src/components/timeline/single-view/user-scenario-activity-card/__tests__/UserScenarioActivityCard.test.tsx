import 'jest';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import UserScenarioActivityCard from '../UserScenarioActivityCard';
import { UserScenarioActivityCardProps } from '../../../../../models/timeline/timeline';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router';

// Return a fixed timestamp when moment().format() is called
jest.mock('moment', () => () => ({format: () => '2019–11–13T12:34:56+00:00'}));

it('renders the UserScenarioActivityCard', () => {
  const props: UserScenarioActivityCardProps = {
    activity: {
      $email_hash: 'string',
      $events: [
        {
          $ts: 4896467488967,
          $event_name: 'event name',
          $properties: 'property',
        },
      ],
      $location: {
        $latlon: ['lat,lon'],
      },
      $origin: {},
      $session_duration: 125646,
      $session_status: 'IN_SESSION',
      $site_id: 'google.com',
      $topics: {},
      $ts: 115165156,
      $ttl: 1212,
      $type: 'type',
      $user_account_id: 'user_account_id',
      $user_agent_id: 'user_agent_id',
      $app_id: 'app_id',
      $node_id: 121555,
      $node_name: 'the node',
      $scenario_name: 'the scenario',
      $scenario_id: 22,
      $previous_node_id: 648787,
      $previous_node_name: 'the prev node',
    },
  };
  const component = TestRenderer.create(
    <IntlProvider locale="en">
      <MemoryRouter>
        <UserScenarioActivityCard {...props} />
      </MemoryRouter>
    </IntlProvider>,
  );
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
