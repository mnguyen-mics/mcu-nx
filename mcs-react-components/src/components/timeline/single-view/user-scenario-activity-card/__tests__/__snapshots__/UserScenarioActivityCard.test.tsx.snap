// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UserScenarioActivityCard 1`] = `
<div
  className="ant-row mcs-card "
  style={Object {}}
>
  <div
    className="ant-row mcs-card-header"
  >
    <div
      className="ant-col-24"
    >
      <span
        className="mcs-card-title"
      >
        Scenario item not found
      </span>
    </div>
  </div>
  <div
    className="ant-row"
    style={Object {}}
  >
    <div
      className="ant-row"
    >
      <div
        className="mcs-card-content-text"
      >
        
      </div>
    </div>
    <div
      className="ant-row border-top sm-footer timed-footer text-right"
    >
      2019–11–13T12:34:56+00:00
    </div>
  </div>
</div>
`;
